services/core-api/app/api/mines/tailings/resources/tailings.py

```diff
@@ -122,3 +122,81 @@ def post(self, mine_guid):
 
         mine.save()
         return mine_tsf, 201
+
+
+class MineTailingsStorageFacilityResource(Resource, UserMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument(
+        'mine_tailings_storage_facility_name',
+        type=str,
+        trim=True,
+        help='Name of the tailings storage facility.',
+        required=True)
+    parser.add_argument(
+        'longitude', type=str, help='Longitude point for the mine.', location='json')
+    parser.add_argument('latitude', type=str, help='Latitude point for the mine.', location='json')
+    parser.add_argument(
+        'consequence_classification_status_code',
+        type=str,
+        trim=True,
+        help='Risk Severity Classification',
+        required=True)
+    parser.add_argument(
+        'tsf_operating_status_code',
+        type=str,
+        trim=True,
+        help='Operating Status of the storage facility',
+        required=True)
+    parser.add_argument(
+        'has_itrb',
+        type=lambda x: True if x.lower() in ("true") else False,
+        trim=True,
+        help='Risk Severity Classification',
+        required=True)
+    parser.add_argument(
+        'eor_party_guid',
+        type=str,
+        help='GUID of the party that is the Engineer of Record for this TSF.',
+        location='json',
+        store_missing=False)
+
+    @api.doc(description='Updates an existing tailing storage facility for the given mine')
+    @requires_role_mine_edit
+    @api.marshal_with(MINE_TSF_MODEL)
+    def put(self, mine_guid, mine_tailings_storage_facility_guid):
+        mine = Mine.find_by_mine_guid(mine_guid)
+
+        if not mine:
+            raise NotFound('Mine not found.')
+
+        mine_tsf = MineTailingsStorageFacility.find_by_tsf_guid(mine_tailings_storage_facility_guid)
+
+        if not mine_tsf:
+            raise NotFound("Tailing Storage Facility not found")
+
+        data = self.parser.parse_args()
+
+        current_app.logger.info(f'Updating {mine_tsf} with {data}')
```

Comment:
Can we remove this logging? I don't think we need this in the logs

Reply:
Done